core/src/main/scala/api/attributes.scala

```diff
@@ -35,15 +39,18 @@ object attributes {
   final class NodeType(`type`: String) extends api.ShowSelf {
     def to_s = `type`
   }
-  val File = new NodeType("file")
-  val Dir  = new NodeType("dir" )
-  val Link = new NodeType("link")
   implicit val _nodeType = new api.Key[NodeType]("type of node")
 
-  final case class Mtime(timestamp: Long)
+  final case class Mtime(timestamp: FileTime) extends FileTimeBased[Mtime](x => Mtime(x)) {
+    override def hashCode = timestamp.toMillis.##
+    override def equals(that: Any) = that match {
+      case Mtime(other) => timestamp.toMillis == other.toMillis
+      case _            => super.equals(that)
```

Comment:
From the `FileTime` docs: 

> equals(Object obj)
> Tests this FileTime for equality with the given object.

with description:

> The result is true if and only if the argument is not null and is a FileTime that represents the same time.

and

> hashCode()
> Computes a hash code for this file time.

So their implementation is not inherited from `Object`